providers/async_registry.py

```
@@ -0,0 +1,310 @@
+                api_key = api_key or ""
+                provider = provider_class(api_key=api_key, base_url=custom_url)
+            else:
+                if not api_key:
+                    return None
+                provider = provider_class(api_key=api_key)
+
+            # Cache the instance
+            instance._async_initialized_providers[provider_type] = provider
+            return provider
+
+        except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)